lib/active_resource/base.rb

```diff
@@ -1644,6 +1649,19 @@ def create_resource_for(resource_name)
         resource
       end
 
+      # Create and return a class definition for a resource whos key began with an integer
+      # or could otherwise not be created normally
+      def create_unknown_resource
+        if self.class.const_defined?(:UnknownResource, false)
+          resource = self.class.const_get(:UnknownResource, false)
+        else
+          resource = self.class.const_set(:UnknownResource, Class.new(ActiveResource::Base))
+        end
+        resource.prefix = self.class.prefix
+        resource.site   = self.class.site
+        resource
+      end
+
```

Comment:
Duplicates resourcing finding from `find_or_create_resource_for` and creation from `create_resource_for`, suggesting that `find_or_create_resource_for` could set `resource_name = "UnknownResource"` as a fallback if the given name isn't a valid Ruby module name.

Reply:
changed this so `find_or_create_resource_for` calls itself with `:UnknownResource` as the name parameter inside the rescue block.